feature-list-streams/src/main/java/com/codandotv/streamplayerapp/feature_list_streams/list/data/ListStreamRepository.kt

```diff
@@ -1,36 +1,47 @@
 package com.codandotv.streamplayerapp.feature_list_streams.list.data
 
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
 import com.codandotv.streamplayerapp.core_networking.handleError.toFlow
-import com.codandotv.streamplayerapp.feature_list_streams.list.domain.model.ListStream
-import com.codandotv.streamplayerapp.feature_list_streams.list.domain.toListStream
+import com.codandotv.streamplayerapp.feature_list_streams.list.domain.model.Genre
+import com.codandotv.streamplayerapp.feature_list_streams.list.domain.model.Stream
+import com.codandotv.streamplayerapp.feature_list_streams.list.domain.toGenres
+import com.codandotv.streamplayerapp.feature_list_streams.list.domain.toStream
 import kotlinx.coroutines.CoroutineDispatcher
-import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.*
 
 interface ListStreamRepository {
-    suspend fun getMovies(): Flow<List<ListStream>>
+    suspend fun getGenres(): Flow<List<Genre>>
+
+    suspend fun topRatedStream(): Flow<Stream>
+
+    fun loadMovies(genre: Genre): Flow<PagingData<Stream>>
 }
 
 class ListStreamRepositoryImpl(
     private val service: ListStreamService,
     private val dispatcher: CoroutineDispatcher,
 ) : ListStreamRepository {
 
-    override suspend fun getMovies(): Flow<List<ListStream>> =
-        service.getGenres()
-            .toFlow()
-            .map {
-                it.genres
-            }
-            .map { genres ->
-                genres.map { genre ->
-                    service.getMovies(genre.id.toString())
-                        .toFlow()
-                        .map {
-                            it.toListStream(
-                                genre.name
-                            )
-                        }.first()
-                }
-            }.flowOn(dispatcher)
+    override suspend fun getGenres(): Flow<List<Genre>> {
+        return service.getGenres().toFlow().map { it.toGenres() }
+    }
+
+    override suspend fun topRatedStream(): Flow<Stream> {
+        return service.getTopRatedMovies().toFlow().map { it.results.first().toStream() }
+    }
+
+    override fun loadMovies(genre: Genre): Flow<PagingData<Stream>> {
+        return Pager(
+            config = PagingConfig(
+                pageSize = 20,
+                maxSize = 500,
+            ),
+            pagingSourceFactory = {
+                StreamDataSource(service, genreName = genre.name, genreId = genre.id)
+            },
+            initialKey = 1
```

Comment:
essa initial key, é algo que incrementa sozinho depois?

Reply:
Sim, no StreamDataSource tem a lógica que faz o incremento. Digamos que esse cara é o ponto de início. Tem algumas APIs que a gente começa a paginação no 0, como essa começamos no 1, eu informei ele como `initialKey`.